### DISCUSSION TITLE
View collections and indirect expand

### DISCUSSION THREAD

**Discussion by sergio9929** [pocketbase/pocketbase#3244] [Q&A] [answered]
following this discussion: #2016

I'm not being able to indirect expand a view collection `listIncidences` with a many to many collection `actionsIncidences`, but I can with the base collection `incidences`. 

## Queries

`actionsIncidences(incidenceId).actionId` works here:
```js
const data = await nuxtApp.$pb.collection('incidences').getList(pagination.value.page, pagination.value.perPage, {
    filter, sort, expand: 'responsibleId.departmentsUsers(userId).departmentId.companyId, secondaryResponsibleId, cause, actionsIncidences(incidenceId).actionId'
})
```

But not here:
```js
const data = await nuxtApp.$pb.collection('listIncidences').getList(pagination.value.page, pagination.value.perPage, {
    filter, sort, expand: 'departmentId, responsibleId, secondaryResponsibleId, cause, actionsIncidences(incidenceId).actionId'
})
```

`responsibleId.departmentsUsers(userId).departmentId.companyId` works in both

## Collections

My base collection `incidences`
```txt
id (id)
name (text)
causeId (single relation)
responsibleId (single relation)
secondaryResponsibleId (single relation)
created (date)
updated (date)
```

My base collection `actionsIncidences`
```txt
id (id)
actionId (single relation)
incidenceId (single relation)
created (date)
updated (date)
```

My view collection `listIncidences`
```sql
SELECT
  incidences.id as id,
  departmentsUsers.departmentId,
  incidences.name,
  incidences.causeId,
  incidences.responsibleId,
  incidences.secondaryResponsibleId,
  incidences.created,
  incidences.updated
FROM incidences
INNER JOIN departmentsUsers
  on departmentsUsers.userId = incidences.responsibleId;
```

**Comment by ganigeorgiev** [comment, marked as answer]
The view is treated as totally different collection  and `actionsIncidences(incidenceId)` targets `incidences`, not `listIncidences`, aka. there is no relation to the view.

To workaround that, you can add the same id column as regular field in your view query and use `incidenceId.actionsIncidences(incidenceId)`.
